### DISCUSSION TITLE
Train model on Pycharm or Visual Code

### DISCUSSION THREAD

**Discussion by TanLuc12** [mrdbourke/pytorch-deep-learning#677] [Q&A] [answered]
Hi there,
I have read section Pytorch going Modular.
How can I train the model within Pycharm or another using this: !python train.py --num_epochs 5 --batch_size 128 --hidden_units 128 --learning_rate 0.003
Because I run this on termianl the error raise:

 !python train.py --num_epochs 5 --batch_size 128 --hidden_units 128 --learning_rate 0.003
!python : The term '!python' is not recognized as the name of a cmdlet, function, script file, or operable
program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
again.
At line:1 char:1
+ !python train.py --num_epochs 5 --batch_size 128 --hidden_units 128 - ...
+ ~~~~~~~
    + CategoryInfo          : ObjectNotFound: (!python:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Thank you very much.

**Comment by Rick-29** [comment, marked as answer]
In your computer you shuld only write `python train.py --num_epochs 5 --batch_size 128 --hidden_units 128 --learning_rate 0.003` without the "!" and in the terminal, the "!" only works in google colab and jupyter notebooks.

**Reply by mrdbourke** [reply]
Hi @TanLuc12 , 

@Rick-29 is right, if you're using the Going Modular scripts outside of a Jupyter/Colab Notebook, you can execute them with `python ...` rather than `!python ...`.

If you're inside a notebook and would like to execute a Python script, you can do so via `!python script_name.py`.
